File: app/src/main/java/com/penseapp/acaocontabilidade/chat/chats/presenter/ChatsReaderPresenter.java
package com.penseapp.acaocontabilidade.chat.chats.presenter;

import com.penseapp.acaocontabilidade.chat.chats.model.Chat;

/**
 * Created by unity on 10/12/16.
 */

public interface ChatsReaderPresenter {
    void subscribeForChatListUpdates();
    void unsubscribeForChatListUpdates();
    void onChatAdded(Chat chat);
    void onChatChanged(Chat chat);
    void onChatRemoved(String chatId);
}

File: app/src/main/java/com/penseapp/acaocontabilidade/chat/chats/presenter/ChatsReaderPresenter.kt
package com.penseapp.acaocontabilidade.chat.chats.presenter

import com.penseapp.acaocontabilidade.chat.chats.model.Chat

/**
 * Created by unity on 10/12/16.
 */

interface ChatsReaderPresenter {

    fun subscribeForChatListUpdates()

    fun unsubscribeForChatListUpdates()

    fun onChatAdded(chat: Chat)

    fun onChatChanged(chat: Chat)

    fun onChatRemoved(chatId: String)
}
